### ISSUE TITLE
Execute plan is not displayed completely

### ISSUE THREAD

**Issue by shaoxiqian** [pingcap/tidb#35687]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

SELECT
  id,
  module_id,
  rel_namespace,
  owned_by,
  created_by,
  updated_by,
  deleted_by,
  created_at,
  updated_at,
  deleted_at
FROM
  (
    SELECT
      crd.id,
      crd.module_id,
      crd.rel_namespace,
      crd.owned_by,
      crd.created_by,
      crd.updated_by,
      crd.deleted_by,
      crd.created_at,
      crd.updated_at,
      crd.deleted_at,
      row_number() over (PARTITION by id) AS pp_rn
    FROM
      compose_record AS crd
    WHERE
      crd.module_id = 288786270480564706
      AND crd.rel_namespace = 288786264809931234
      AND crd.deleted_at IS NULL
    ORDER BY
      created_at DESC,
      id DESC
  ) AS base
WHERE
  pp_rn = 1
LIMIT
  21 
### 2. What did you expect to see? (Required)
A complete execution plan like this ⬇️
![image](https://user-images.githubusercontent.com/85105033/175278184-63de8f85-4630-4c86-bc20-c540219759c1.png)

### 3. What did you see instead (Required)
![image](https://user-images.githubusercontent.com/85105033/175277761-529a2c6f-e4ec-4c60-bc69-be7ecae062e3.png)

### 4. What is your TiDB version? (Required)

nightly
mysql -h 172.16.6.16 -P30041 -uroot -D xxx
Dashboard: http://172.16.6.16:32380/dashboard/

**Comment by time-and-fate** [comment]
```sql
use test;
create table t (a int, b int);
select row_number() over (partition by a) from t;
select plan from information_schema.statements_summary where SAMPLE_USER = 'root' and stmt_type = 'Select' and SCHEMA_NAME = 'test' and PLAN like '%Window%';
```

Expected:
The full plan tree is displayed.
```
       id                              task            estRows operator info                                                                                   actRows execution info                                                                                                                                                                            memory          disk
        Projection_7                    root            10000   Column#5                                                                                        0       time:3.06ms, loops:1, Concurrency:5                                                                                                                                                       5.57 KB         N/A
        └─Shuffle_12                    root            10000   execution info: concurrency:5, data sources:[TableReader_10]                                    0       time:3.03ms, loops:1, ShuffleConcurrency:5                                                                                                                                                N/A             N/A
          └─Window_8                    root            10000   row_number()->Column#5 over(partition by test.t.a rows between current row and current row)     0       time:14.6ms, loops:5                                                                                                                                                                      N/A             N/A
            └─Sort_11                   root            10000   test.t.a                                                                                        0       time:14.6ms, loops:5                                                                                                                                                                      0 Bytes         0 Bytes
              └─ShuffleReceiver_13      root            10000                                                                                                   0       time:14.6ms, loops:5                                                                                                                                                                      N/A             N/A
                └─TableReader_10        root            10000   data:TableFullScan_9                                                                            0       time:2.95ms, loops:1, cop_task: {num: 1, max: 226.1µs, proc_keys: 0, rpc_num: 2, rpc_time: 891.5µs, copr_cache_hit_ratio: 0.00, distsql_concurrency: 15}, backoff{regionMiss: 2ms}        184 Bytes       N/A
                  └─TableFullScan_9     cop[tikv]       10000   table:t, keep order:false, stats:pseudo                                                         0       tikv_task:{time:0s, loops:1}, scan_detail: {total_keys: 1, rocksdb: {block: {}}}                                                                                                          N/A             N/A 
```

Actual:
The children of `ShuffleReceiver` is missing.
```
       id                              task    estRows operator info                                                                                   actRows execution info                                   memory  disk
        Projection_7                    root    10000   Column#5                                                                                        0       time:383µs, loops:1, Concurrency:5               5.57 KB N/A
        └─Shuffle_12                    root    10000   execution info: concurrency:5, data sources:[TableReader_10]                                    0       time:364µs, loops:1, ShuffleConcurrency:5        N/A     N/A
          └─Window_8                    root    10000   row_number()->Column#5 over(partition by test.t.a rows between current row and current row)     0       time:1.61ms, loops:5                             N/A     N/A
            └─Sort_11                   root    10000   test.t.a                                                                                        0       time:1.56ms, loops:5                             0 Bytes 0 Bytes
              └─ShuffleReceiver_17      root    10000                                                                                                   0                                                        N/A     N/A 
```

**Comment by time-and-fate** [comment]
This issue has been fixed from v6.2.0 in https://github.com/pingcap/tidb/pull/36069 (one of several PRs in related refactor work).
We will create another PR to fix it on release-6.1.

**Comment by time-and-fate** [comment]
Fix for release-6.1: #40475
Expected to be released in v6.1.4
